src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -2533,6 +2818,95 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
+                enabled=True)
+            addons.remove('azure-policy')
+        if 'virtual-node' in addons:
+            aci_subnet_name = self.context.get_aci_subnet_name()
+            vnet_subnet_id = self.context.get_vnet_subnet_id()
+            if not aci_subnet_name or not vnet_subnet_id:
+                raise RequiredArgumentMissingError(
+                    '"--enable-addons virtual-node" requires "--aci-subnet-name" and "--vnet-subnet-id".')
+            # TODO: how about aciConnectorwindows, what is its addon name?
+            os_type = self.context.get_virtual_node_addon_os_type()
+            addon_profiles[CONST_VIRTUAL_NODE_ADDON_NAME + os_type] = ManagedClusterAddonProfile(
+                enabled=True,
+                config={CONST_VIRTUAL_NODE_SUBNET_NAME: aci_subnet_name}
+            )
+            # set intermediate
+            self.context.set_intermediate("enable_virtual_node", True, overwrite_exists=True)
+            addons.remove('virtual-node')
+        if 'ingress-appgw' in addons:
+            addon_profile = ManagedClusterAddonProfile(enabled=True, config={})
+            appgw_name = self.context.get_appgw_name()
+            appgw_subnet_cidr = self.context.get_appgw_subnet_cidr()
+            appgw_id = self.context.get_appgw_id()
+            appgw_subnet_id = self.context.get_appgw_subnet_id()
+            appgw_watch_namespace = self.context.get_appgw_watch_namespace()
+            if appgw_name is not None:
+                addon_profile.config[CONST_INGRESS_APPGW_APPLICATION_GATEWAY_NAME] = appgw_name
+            if appgw_subnet_cidr is not None:
+                addon_profile.config[CONST_INGRESS_APPGW_SUBNET_CIDR] = appgw_subnet_cidr
+            if appgw_id is not None:
+                addon_profile.config[CONST_INGRESS_APPGW_APPLICATION_GATEWAY_ID] = appgw_id
+            if appgw_subnet_id is not None:
+                addon_profile.config[CONST_INGRESS_APPGW_SUBNET_ID] = appgw_subnet_id
+            if appgw_watch_namespace is not None:
+                addon_profile.config[CONST_INGRESS_APPGW_WATCH_NAMESPACE] = appgw_watch_namespace
+            addon_profiles[CONST_INGRESS_APPGW_ADDON_NAME] = addon_profile
+            # set intermediate
+            self.context.set_intermediate("ingress_appgw_addon_enabled", True, overwrite_exists=True)
+            addons.remove('ingress-appgw')
+        if 'confcom' in addons:
+            addon_profile = ManagedClusterAddonProfile(
+                enabled=True, config={CONST_ACC_SGX_QUOTE_HELPER_ENABLED: "false"})
+            if self.context.get_enable_sgxquotehelper():
+                addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
+            addon_profiles[CONST_CONFCOM_ADDON_NAME] = addon_profile
+            addons.remove('confcom')
+        # error out if any (unrecognized) addons remain
+        if addons:
+            raise InvalidArgumentValueError('"{}" {} not recognized by the --enable-addons argument.'.format(
```

Comment:
I think we should put this validation to ealier stage.

because when we get the workspace, it will try to create something, but then the operation failed.
it's not acceptable by our customer.

Reply:
Sure, but this would change the original behavior of this command. Changed in commit [8838244](https://github.com/Azure/azure-cli/pull/19478/commits/883824466ad66457076df428cd7da1234578cd17).